src/Oscoin/Protocol/Sync/RealWorld.hs

```diff
@@ -239,9 +242,19 @@ syncNode
        , Ord (Beneficiary c)
        , Buildable (Hash c)
        )
-    => SyncT c tx s m ()
-syncNode = forever $ do
-    sync `catchError` \(e :: SyncError) ->
-       recordEvent $ Telemetry.Events.NodeSyncError (toException e)
-    -- Throttle each iteration of the algorithm by 30 seconds.
-    liftIO $ threadDelay 30000000
+    => (Either SyncError (LocalTip c tx s, RemoteTip c tx s) -> m ())
+    -- ^ A callback which will fire once the sync has been performed the
+    -- first time.
+    -> SyncT c tx s m ()
+syncNode onReady = do
+    lift . onReady =<< catchingSyncErrors (syncUntil (\_ _ _ -> False))
```

Comment:
Why don’t we use `isDone` here?

Reply:
The intention was to make sure `syncUntil` was actually fetching some blocks _no matter what_, but I think you are right, I think calling `isDone` should be perfectly fine here.

Reply:
8c9272c5